### ISSUE TITLE
Can't query the joint position in more than 30Hz.

### ISSUE THREAD

**Issue by hang0610** [leap-hand/LEAP_Hand_API#8]
Hi,

Here is my situation. I modified the ros Rate from 30Hz to 60Hz in the script `ros_example.py`, then I found the rate of the publishing topic "/leaphand_node/cmd_leap" was still around 30Hz. I tested the query time of "leap_position" which turned out to be around 0.03s\~0.04s. But according to your document, the query time of the single "leap_position" should be 0.01s\~0.02s (90Hz)?

How can I query the single "leap_position" in a higher rate than 30Hz?

**Comment by hang0610** [comment]
Update: 

If I uninstall other USB devices in my MiniPC and only keep the leap-hand USB, the query rate can increase to around 45Hz. Is the query rate lower than the expected because of the limitation of Micro-B (USB-2.0) or my MiniPC?

**Comment by Lucky7Chess** [comment]
I have also run into this issue.  It is an issue with some computers drivers and the u2d2.  You can try switching controllers or drivers or baud rate but there's no pattern I could find that leads to this.  You can also contact Dynamixel support and maybe they can help you debug further.

**Comment by hang0610** [comment]
> I have also run into this issue. It is an issue with some computers drivers and the u2d2. You can try switching controllers or drivers or baud rate but there's no pattern I could find that leads to this. You can also contact Dynamixel support and maybe they can help you debug further.

Got it. Thanks Kenny! I think there is a need to keep this issue open to see if anyone can provide a specific solution in the future.

**Comment by happygaoxiao** [comment]
With ROS2 we can get the joint states at 60Hz. See an eample from the official [repo](https://github.com/dynamixel-community/dynamixel_hardware)

**Comment by Lucky7Chess** [comment]
Interesting, maybe the ros2 API is better than the python one for Dynamixel....not sure because it's super computer dependent.  I have also talked to Dynamixel a lot about it but it seems to be hard issue to diagnose.

**Comment by Lucky7Chess** [comment]
Few things I have noticed.  One I changed from bulkread to syncread which speeds things up a little bit.  Two, when you read multiple of position, velocity, effort etc, use the combined function queries that exist, do not pull them individually.  Also make sure the driver is set correctly if you're using Ubuntu.  With all that combined I can command and then query at 90hz using my computer easily without any issues.
